src/main/scala/uk/gov/ons/registers/model/SelectionTypes.scala

```diff
@@ -5,9 +5,9 @@ import scala.util.Try
 sealed trait SelectionTypes
 
 object SelectionTypes {
-  case object Census extends SelectionTypes
-  case object PrnSampling extends SelectionTypes
-  case object Universal extends SelectionTypes
+  case class Census() extends SelectionTypes
```

Comment:
I'm surprised this isn't generating lots of warnings.  Case classes with no values are frowned upon.
Your original use of case objects seem more appropriate.

Reply:
Good spot, this was the type class - I'll discuss this with you.